cclib/parser/molcasparser.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of cclib (http://cclib.github.io), a library for parsing
+# and interpreting the results of computational chemistry packages.
+#
+# Copyright (C) 2014-2015, the cclib development team
+#
+# The library is free software, distributed under the terms of
+# the GNU Lesser General Public version 2.1 or later. You should have
+# received a copy of the license along with cclib. You can also access
+# the full license online at http://www.gnu.org/copyleft/lgpl.html.
+
+"""Parser for Molcas output files"""
+
+from __future__ import print_function
+
+import re
+
+from . import logfileparser
+from . import utils
+
+
+class Molcas(logfileparser.Logfile):
+    """A Molcas log file."""
+
+    def __init__(self, *args, **kwargs):
+
+        # Call the __init__ method of the superclass
+        super(Molcas, self).__init__(logname="Molcas", *args, **kwargs)
+
+    def __str__(self):
+        """Return a string representation of the object."""
+        return "Molcas log file %s" % (self.filename)
+
+    def __repr__(self):
+        """Return a representation of the object."""
+        return 'Molcas("%s")' % (self.filename)
+
+    def normalisesym(self, label):
+        """Does Molcas require symmetry label normalization?"""
+
+    def before_parsing(self):
+        # Compile the regex for extracting the element symbol from the
+        # atom label in the "Molecular structure info" block.
+        self.re_atomelement = re.compile('([a-zA-Z]+)\d+')   
+
+        # Compile the dashes-and-or-spaces-only regex.
+        self.re_dashes_and_spaces = re.compile('^[\s-]+$')
```

Comment:
There is minimal advantage to compiling regexes. Python internally compiles and caches regexes, so unless you have a huge number of them, explicitly compiling is unlikely to help. https://stackoverflow.com/questions/452104/is-it-worth-using-pythons-re-compile